sentry-core/src/profiling.rs

```diff
@@ -171,72 +169,104 @@ pub fn debug_images() -> Vec<DebugImage> {
 
+        stacks.push(stack);
         samples.push(Sample {
-            frames,
-            thread_name: String::from_utf8_lossy(&sample.thread_name[0..sample.thread_name_length])
-                .into_owned(),
+            stack_id: (stacks.len() - 1) as u32,
             thread_id: sample.thread_id,
-            nanos_relative_to_start: sample
+            relative_timestamp_ns: sample
                 .sample_timestamp
                 .duration_since(rep.timing.start_time)
                 .unwrap()
                 .as_nanos() as u64,
         });
+
+        thread_metadata
+            .entry(sample.thread_id.to_string())
+            .or_insert(ThreadMetadata {
+                name: Some(
+                    String::from_utf8_lossy(&sample.thread_name[0..sample.thread_name_length])
+                        .into_owned(),
+                ),
+            });
     }
-    let sampled_profile = SampledProfile {
-        start_time_nanos: rep
-            .timing
-            .start_time
-            .duration_since(SystemTime::UNIX_EPOCH)
-            .unwrap()
-            .as_nanos() as u64,
-        start_time_secs: rep
-            .timing
-            .start_time
-            .duration_since(SystemTime::UNIX_EPOCH)
-            .unwrap()
-            .as_secs(),
-        duration_nanos: rep.timing.duration.as_nanos() as u64,
-        samples,
-    };
 
-    let profile: Profile = Profile {
-        duration_ns: sampled_profile.duration_nanos,
-        debug_meta: DebugMeta {
+    SampleProfile {
+        version: Version::V1,
+        debug_meta: Some(DebugMeta {
             sdk_info: None,
             images: debug_images(),
+        }),
+        device: DeviceMetadata {
+            architecture: Some(std::env::consts::ARCH.to_string()),
+        },
+        os: OSMetadata {
+            name: sys_info::os_type().unwrap(),
+            version: sys_info::os_release().unwrap(),
+            build_number: None,
+        },
+        runtime: Some(RuntimeMetadata {
+            name: "rustc".to_string(),
+            version: rustc_version_runtime::version().to_string(),
+        }),
+        environment: match &transaction.environment {
+            Some(env) => env.to_string(),
+            _ => "".to_string(),
         },
-        platform: "rust".to_string(),
-        architecture: Some(std::env::consts::ARCH.to_string()),
-        trace_id,
-        transaction_name,
-        transaction_id,
-        profile_id: uuid::Uuid::new_v4(),
-        sampled_profile,
-        os_name: sys_info::os_type().unwrap(),
-        os_version: sys_info::os_release().unwrap(),
-        version_name: env!("CARGO_PKG_VERSION").to_string(),
-        version_code: build_id::get().to_simple().to_string(),
-    };
 
-    profile
+        event_id: uuid::Uuid::new_v4(), // double check this
+        release: format!(
+            "{} ({})",
+            env!("CARGO_PKG_VERSION"),
+            build_id::get().to_simple(),
+        ),
```

Comment:
I believe `CARGO_PKG_VERSION` would be the version of the `sentry-core` crate in this case. Is that intentional or should this rather use the release as configured?

Reply:
we're moving to sentry SDK release. Good 👍🏻